src/functional/iterator/embedded.py

```diff
@@ -586,11 +606,28 @@ def shape(self):
         return self.array().shape
 
 
+def _is_tuple_axis(axis: Axis):
+    return axis is None
+
+
 def get_ordered_indices(
-    axes: Iterable[Axis], pos: Mapping[str, FieldIndex]
+    axes: Iterable[Axis], pos: Mapping[str, FieldIndex | SparsePositionEntry]
 ) -> tuple[FieldIndex, ...]:
-    assert all(axis.value in [*pos.keys()] for axis in axes)
-    return tuple(pos[axis.value] for axis in axes)
+    res = list[FieldIndex]()
+    pos = deepcopy(pos)
+    for axis in axes:
+        if _is_tuple_axis(axis):
+            res.append(slice(None))
+        else:
+            assert axis is not None
+            assert axis.value in pos
+            elem = pos[axis.value]
+            if isinstance(elem, list):
+                res.append(elem.pop(0))
```

Comment:
Do you mean ```res.append(elem[0])```?

Reply:
No, it's correct, but super ugly. What I do by popping is consuming the next element in the list of sparse dimensions (therefore the deepcopy in the beginning).

Reply:
Maybe adding this explanation as a comment would help.

Reply:
added a comment